server/webhook.go

```diff
@@ -368,40 +369,73 @@ func (s *Server) processFile(ctx context.Context, pushEvent vcs.PushEvent, repo
+					creatorID := s.getIssueCreatorID(ctx, pushEvent.FileCommit.AuthorEmail)
+					if err := s.createIssueFromMigrationDetailList(ctx, issueName, pushEvent, creatorID, repo.ProjectID, fileInfo.migrationInfo.Type, migrationDetailListForFile); err != nil {
+						return "", false, activityCreateList, echo.NewHTTPError(http.StatusInternalServerError, "Failed to create issue").SetInternal(err)
+					}
+					createdIssueList = append(createdIssueList, issueName)
+				}
+			} else {
+				log.Debug("Ignored schema file for non-SDL project", zap.String("fileName", fileInfo.item.FileName), zap.String("type", string(fileInfo.item.ItemType)))
+			}
+		} else { // fileInfo.fType == migrationFileType
+			// This is a migration-based DDL or DML file and we would allow it for both DDL and SDL schema change type project.
+			// For DDL schema change type project, this is expected.
+			// For SDL schema change type project, we allow it because:
+			// 1) DML is always migration-based.
+			// 2) We may have a limitation in SDL implementation.
+			// 3) User just wants to break the glass.
+			if fileInfo.migrationInfo.Type == db.Migrate {
+				migrationTypeDDL = db.Migrate
+			}
+			migrationDetailListForFile, activityCreateListForFile := s.prepareIssueFromPushEventDDL(ctx, repo, pushEvent, fileInfo.item.FileName, fileInfo.item.ItemType, fileInfo.migrationInfo)
+			activityCreateList = append(activityCreateList, activityCreateListForFile...)
+			migrationDetailList = append(migrationDetailList, migrationDetailListForFile...)
+		}
 	}
 
 	if len(migrationDetailList) == 0 {
-		return "", false, activityCreateList, nil
+		return "", len(createdIssueList) != 0, activityCreateList, nil
 	}
 
-	// Create schema update issue
-	issueName := fmt.Sprintf("%s by %s", migrationDescription, strings.TrimPrefix(file, repo.BaseDirectory+"/"))
+	// Create one issue per push event for DDL project, or non-schema files for SDL project.
+	issueName := pushEvent.CommitList[0].Title
 	creatorID := s.getIssueCreatorID(ctx, pushEvent.FileCommit.AuthorEmail)
-	if err := s.createIssueFromMigrationFile(ctx, issueName, pushEvent, creatorID, repo.ProjectID, migrationInfo.Type, migrationDetailList); err != nil {
-		return "", false, activityCreateList, echo.NewHTTPError(http.StatusInternalServerError, "Failed to create issue").SetInternal(err)
+	if err := s.createIssueFromMigrationDetailList(ctx, issueName, pushEvent, creatorID, repo.ProjectID, migrationTypeDDL, migrationDetailList); err != nil {
+		return "", len(createdIssueList) != 0, activityCreateList, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to create issue %s", issueName)).SetInternal(err)
 	}
 
-	return fmt.Sprintf("Created issue %q from file %q", issueName, file), true, activityCreateList, nil
+	return fmt.Sprintf("Created issue %q from push event", strings.Join(createdIssueList, ",")), true, activityCreateList, nil
+}
+
+func sortFilesBySchemaVersionGroupByDatabase(fileInfoList []fileInfo) []fileInfo {
```

Comment:
sortFilesBySchemaVersionGroupByDatabase, the name seems to be doing the right thing that we should create different issues for different database groups. Changes for the same database should belong to the same issue.

Reply:
Done.